aws_cloudtrail_rules/aws_iam_user_recon_denied.py

```diff
@@ -0,0 +1,54 @@
+from fnmatch import fnmatch
+from ipaddress import ip_address
+from panther_oss_helpers import check_threshold  # pylint: disable=import-error
+
+# service/event patterns to monitor
+RECON_ACTIONS = {
+    'dynamodb': ['List*', 'Describe*', 'Get*'],
+    'ec2': ['Describe*', 'Get*'],
+    'iam': ['List*', 'Get*'],
+    's3': ['List*', 'Get*'],
+    'rds': ['Describe*', 'List*'],
+}
+THRESH = 5
+THRESH_TTL = 600  # 10 minutes
+
+
+def rule(event):
+    # Filter events
+    if event['errorCode'] != 'AccessDenied':
```

Comment:
Is errorCode always defined? `if event.get('errorCode') != 'AccessDenied'` to be safe

Reply:
Good call

Reply:
errorCode is not a required field, should be a safe lookup.